exercises/practice/square-root/.meta/example.sml

```diff
@@ -0,0 +1,12 @@
+fun squareRoot (radicand: int): int =
+  let
+    fun recurse guess =
+      let
+        val estimate = (guess * (guess + 1) + radicand) div (2 * guess)
```

Comment:
Can you provide a brief description or reference for the methodology which is used to compute the square root (as a docstring)? This looks a bit like [Heron's method](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Heron's_method), but possibly slightly adapted to the integer world?

Reply:
Comment added. We implement Heron's method using integer arithmetic, rounding the estimate to the nearest natural number.

Reply:
OK sounds reasonable. I didn't rigorously check it though, the tests pass and I assume that you made sure some kind of correctness in general?

If yes I will merge it.

Reply:
I have tested radicand values 1 .. 10000

My V [implementation](https://github.com/exercism/vlang/blob/main/exercises/practice/square-root/.meta/example.v) also works for 0, but that isn't important as the instructions exclude 0.